dbms/src/Flash/Mpp/ReceivedMessageQueue.h

```diff
@@ -36,59 +58,67 @@ enum class ReceiverMode
 
 class ReceivedMessageQueue
 {
-    /// msg_channel is a bounded queue that saves the received messages
-    /// msg_channels_for_fine_grained_shuffle is unbounded queues that saves fine grained received messages
-    /// all the received messages in msg_channels_for_fine_grained_shuffle must be saved in msg_channel first, so the
-    /// total size of `ReceivedMessageQueue` is still under control even if msg_channels_for_fine_grained_shuffle
-    /// is unbounded queues
-    /// for non fine grained shuffle, all the read/write to the queue is based on msg_channel/grpc_recv_queue
-    /// for fine grained shuffle
-    /// write: the writer first write the msg to msg_channel/grpc_recv_queue, if write success, then write msg to msg_channels_for_fine_grained_shuffle
-    /// read: the reader read msg from msg_channels_for_fine_grained_shuffle, and reduce the `remaining_consumers` in msg, if `remaining_consumers` is 0, then
-    ///       remove the msg from msg_channel/grpc_recv_queue
-    std::vector<MsgChannelPtr> msg_channels_for_fine_grained_shuffle;
-    MsgChannelPtr msg_channel;
-    std::shared_ptr<GRPCReceiveQueue<ReceivedMessagePtr>> grpc_recv_queue;
-    size_t fine_grained_channel_size;
-    LoggerPtr log;
-
 public:
+    ReceivedMessageQueue(
+        const CapacityLimits & queue_limits,
+        const LoggerPtr & log_,
+        std::atomic<Int64> * data_size_in_queue_,
+        bool enable_fine_grained,
+        size_t fine_grained_channel_size_);
+
     template <bool need_wait>
     MPMCQueueResult pop(size_t stream_id, ReceivedMessagePtr & recv_msg);
 
     template <bool is_force>
-    bool pushToMessageChannel(ReceivedMessagePtr & received_message, ReceiverMode mode);
+    bool pushLocalPacket(size_t source_index,
+                         const String & req_info,
+                         const TrackedMppDataPacketPtr & tracked_packet,
+                         ReceiverMode mode);
 
-    GRPCReceiveQueueRes pushToGRPCReceiveQueue(ReceivedMessagePtr & received_message);
-
-    ReceivedMessageQueue(
-        const AsyncRequestHandlerWaitQueuePtr & conn_wait_queue,
-        const LoggerPtr & log_,
-        const CapacityLimits & queue_limits,
-        bool enable_fine_grained,
-        size_t fine_grained_channel_size_);
+    MPMCQueueResult pushGRPCPacket(size_t source_index,
```

Comment:
I found that `pushLocalPacket` and `pushGRPCPacket` is not very accurate because in my understanding:
local tunnel, sync grpc receiver, async grpc receiver version 1 will use `pushLocalPacket`, and async grpc receiver version 2 will use `pushGRPCPacket`?

Reply:
Yes. I think sync grpc receiver and async grpc version 1 are the legacy and should be removed soon.

Reply:
async grpc version 1 will be removed, but sync grpc maybe kept for debug purpose.